### PR TITLE
Feature/user data handlers

### LINKED ISSUE #62: Add handlers for the user data
- [x] GET for own user data
- [x] PATCH for own user data
- [x] GET for others user data (for admin)
- [x] PATCH for others user data (for admin)

@riesinger Should there also be POST and/or DELETE handlers for signing up/deleting the account?

### PR DESCRIPTION
**Pull request by TheHamloer**
Closes #62 
For deleting a user and creating users as an admin we probably should open new issues...

### CODE REVIEW

**Comment by TheHamloer** [comment]
I just noticed that updating with an empty body resets all fields because the JSONDecoder fills everything with the default value and does the update with that...

**Comment by TheHamloer** [comment]
The problem described above is fixes now so the PR is reopened

File: auth/auth.go [hunk 1]
```diff
@@ -120,3 +120,64 @@ func GetUserByID(uid int) (*models.User, error) {
...
 	return sProvider.RemoveSession(sess)
 }
+
+func UpdateUser(uid int, updates map[string]interface{}) (user *models.User, err error) {
+	user, err = GetUserByID(uid)
+	if err != nil {
+		return
+	}
+
+	if email, ok := updates["email"]; ok == true {
```
Comment (riesinger) [inline auth/auth.go]: Actually, you don't need the `== true` :wink:

File: auth/auth.go [hunk 2]
```diff
@@ -120,3 +120,64 @@ func GetUserByID(uid int) (*models.User, error) {
 func RemoveSession(sess models.Session) (err error) {
 	return sProvider.RemoveSession(sess)
 }
+
+func UpdateUser(uid int, updates map[string]interface{}) (user *models.User, err error) {
```
Comment (TheHamloer) [inline auth/auth.go]: Any suggestions on how to make this function prettier are very much welcome 😅

Comment (riesinger) [reply auth/auth.go]: Hm :thinking: I got none right now, but I'll think about it...

File: auth/auth.go (hunk 1 above)
Comment (TheHamloer) [reply auth/auth.go]: I know but it makes it a little bit more readable I think 😄

File: auth/auth.go (hunk 2 above)
Comment (TheHamloer) [reply auth/auth.go]: 👍

**Pull request merged**